server/src/main/java/io/littlehorse/server/streamsimpl/ServerTopology.java

```diff
@@ -67,45 +69,88 @@ public class ServerTopology {
     public static final String CORE_REPARTITION_PROCESSOR =
         "core-repartition-processor";
 
-    public static final String GLOBAL_META_SOURCE = "global-metadata-cl-source";
+    public static final String GLOBAL_METADATA_SOURCE = "global-metadata-cl-source";
     public static final String GLOBAL_STORE = "global-metadata-store";
-    public static final String GLOBAL_META_PROCESSOR = "global-metadata-processor";
+    public static final String GLOBAL_METADATA_PROCESSOR =
+        "global-metadata-processor";
+
+    public static final String METADATA_SOURCE = "metadata-source";
+
+    public static final String METADATA_PROCESSOR = "metadata-processor";
+
+    public static final String METADATA_STORE = "metadata-store";
+
+    public static final String GLOBAL_METADATA_SINK = "global-metadata-sink";
 
     public static Topology initCoreTopology(
         LHConfig config,
         KafkaStreamsServerImpl server
     ) {
         Topology topo = new Topology();
         WfSpecCache wfSpecCache = new WfSpecCache();
+        Serializer<Object> sinkValueSerializer = (topic, output) -> {
+            CommandProcessorOutput cpo = (CommandProcessorOutput) output;
+            if (cpo.payload == null) {
+                return null;
+            }
+
+            return cpo.payload.toBytes(config);
+        };
+
+        TopicNameExtractor<String, Object> sinkTopicNameExtractor = (
+                key,
+                coreServerOutput,
+                ctx
+            ) ->
+            ((CommandProcessorOutput) coreServerOutput).topic;
+
         topo.addSource(
             CORE_SOURCE, // source name
             Serdes.String().deserializer(), // key deserializer
             new LHDeserializer<>(Command.class, config), // value deserializer
             config.getCoreCmdTopicName() // source topic
         );
 
+        topo.addSource(
+            METADATA_SOURCE, // source name
+            Serdes.String().deserializer(), // key deserializer
+            new LHDeserializer<>(Command.class, config), // value deserializer
+            config.getMetadataCmdTopicName() // source topic
+        );
+
+        topo.addProcessor(
+            METADATA_PROCESSOR,
+            () ->
```

Comment:
I am regretting the decision to use `prettier`

Reply:
Hahahaha I was thinking the same. I think we need to configure it better :)